src/boot/bootctl.c

```diff
@@ -1068,23 +1059,22 @@ static int verb_status(int argc, char *argv[], void *userdata) {
                         printf("   Setup Mode: %s\n", r ? "setup" : "user");
                 printf("\n");
 
-                printf("Loader:\n");
+                printf("Current Loader:\n");
                 printf("      Product: %s\n", strna(loader));
                 if (!sd_id128_is_null(loader_part_uuid))
-                        printf("    Partition: /dev/disk/by-partuuid/%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x\n",
+                        printf("          ESP: /dev/disk/by-partuuid/%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x\n",
                                SD_ID128_FORMAT_VAL(loader_part_uuid));
                 else
-                        printf("    Partition: n/a\n");
+                        printf("          ESP: n/a\n");
                 printf("         File: %s%s\n", special_glyph(TREE_RIGHT), strna(loader_path));
                 printf("\n");
         } else
-                printf("System:\n    Not booted with EFI\n");
+                printf("System:\n    Not booted with EFI\n\n");
 
-        r = status_binaries(arg_path, uuid);
-        if (r < 0)
-                return r;
+        if (arg_path)
+                r = status_binaries(arg_path, uuid);
```

Comment:
So you stop checking `r` and continue, even if there are errors. I think this is OK, because it is better to try to print as much information as possible, even if something fails early in the process. But at the end, failure should be returned in any of the parts failed. So I think this should be changed to:
```C
r = find_esp(...);
if (arg_path)
     r2 = status_binaries(...);
if (is_efi_boot())
     r3 = status_variables();
return r < 0 ? r : r2 < 0 ? r2 : r3;
```